web/src/main/java/org/springframework/security/web/util/matcher/AntPathRequestMatcher.java

```diff
@@ -159,9 +160,12 @@ public boolean matches(HttpServletRequest request) {
 
 	@Override
 	public MatchResult matcher(HttpServletRequest request) {
-		if (this.matcher == null || !matches(request)) {
+		if (!matches(request)) {
```

Comment:
It seems like `this.matcher == null` should be the first check in order to avoid potential NPEs in `matches(HttpServletRequest)`.

Reply:
The `matcher` will only be null if the `pattern` is `MATCH_ALL`, and we need to check the `HttpMethod` first.

Reply:
Good point. Is the null check necessary, then? It seems like it could be removed.

Reply:
The null check is necessary since `matcher` is used to extract variables, for the `MATCH_ALL` pattern there will be an empty `Map` of variables.